solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -292,7 +346,9 @@ public boolean isReadOnly() {
 
   @Override
   public String toString() {
-    return "DocCollection("+name+"/" + znode + "/" + znodeVersion + ")=" + toJSONString(this);
+    return "DocCollection("+name+"/" + znode + "/" + znodeVersion
+        + (perReplicaStates == null ? "": perReplicaStates.toString())+")="
```

Comment:
Do we want to add a space after the znodeVersion in the output if perReplicaStates is non null?